allensdk/api/cloud_cache/cloud_cache.py

```diff
@@ -0,0 +1,525 @@
+import os
+import copy
+import io
+import pathlib
+import pandas as pd
+from typing import TypedDict
+import boto3
+from botocore import UNSIGNED
+from botocore.client import Config
+from allensdk.internal.core.lims_utilities import safe_system_path
+from allensdk.api.cloud_cache.manifest import Manifest
+from allensdk.api.cloud_cache.file_attributes import CacheFileAttributes  # noqa: E501
+from allensdk.api.cloud_cache.utils import file_hash_from_path  # noqa: E501
+from allensdk.api.cloud_cache.utils import bucket_name_from_url  # noqa: E501
+from allensdk.api.cloud_cache.utils import relative_path_from_url  # noqa: E501
+
+
+class LocalFileDescription(TypedDict):
+    local_path: pathlib.Path
+    exists: bool
+    file_attributes: CacheFileAttributes
+
+
+class CloudCache(object):
+    """
+    A class to handle the downloading and accessing of data served from a cloud
+    storage system
+
+    Parameters
+    ----------
+    cache_dir: str or pathlib.Path
+        Path to the directory where data will be stored on the local system
+
+    ***** THIS IS JUST A BASE CLASS AND CANNOT BE INSTANTIATED *****
+
+    Actual implementations of this class must implement
+    ===================================================
+
+    def _list_all_manifests(self) -> list:
+        Return a list of all of the file names of the manifests associated
+        with this dataset
+
+    def _download_manifest(self,
```

Comment:
and then, these would have the `@abstractmethod` decorator on them and/or raise `NotImplementedError`

Reply:
oops, i am commenting your docstring